Compilation fails with syn = 1.0.7

rtic-syntax depends on `syn`  = `"1"`, but compiling it with 1.0.7 fails:

```
error[E0599]: no method named `span` found for struct `ParseBuffer<'_>` in the current scope
   --> src/parse.rs:381:25
    |
381 |                 content.span(),
    |                         ^^^^ method not found in `ParseBuffer<'_>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `rtic-syntax`
```

bumping syn to 1.0.67 fixes it, but I guess min version should be updated accordingly.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ quote = "1"
 
 [dependencies.syn]
 features = ["extra-traits", "full"]
-version = "1"
+version = "1.0.12"
 
 [dev-dependencies]
 mock = { path = "mock" }

